### PR TITLE
OSAC 170 - Database issue: DllNotFoundException: Unable to load shared library 'SNI.dll'

### PR DESCRIPTION
**Pull request by alcxs**
Description
Database issue for Net5: DllNotFoundException: Unable to load shared library 'SNI.dll'
How to test
Load Database activity pack and test
Jira
[OSAC-170](https://uipath.atlassian.net/browse/OSAC-170)
Label:
24S

### CODE REVIEW

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+        private static extern IntPtr LoadLibrary(string libname);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto)]
+        private static extern bool FreeLibrary(IntPtr hModule);
+
+
+        public static void SNILoadWorkaround()
+        {
+#if NETCOREAPP
```
Comment (ovidiuiliescu) [inline Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: These can be unified, just the wix-x64/x86 is different between them.

Same for the other instance of this class.

**Review by ovidiuiliescu — CHANGES_REQUESTED** [review]

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            IntPtr Handle = LoadLibrary(Path.GetFullPath(typeof(DataConnectionDialog).Assembly.Location + @"\..\runtimes\win-x64\native\sni.dll"));
+            if (Handle == IntPtr.Zero)
+            {
+                int errorCode = Marshal.GetLastWin32Error();
+                throw new Exception(string.Format("Failed to load library (ErrorCode: {0})", errorCode));
+            }
+#endif
+
+#if NETFRAMEWORK
+            IntPtr Handle = LoadLibrary(Path.GetFullPath(typeof(DataConnectionDialog).Assembly.Location + @"\..\runtimes\win-x86\native\sni.dll"));
```
Comment (ovidiuiliescu) [inline Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: In practice, can we reference the same type here as in the other class (or vice-versa) ?

If so, we can just have one file containing this class and have it added in the other place as a linked file - this results in less duplication.

Also, not entirely sure why we need this class duplicated in the first place.

Comment (alcxs) [reply Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: added class as linked file to remove the duplication

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs (hunk 1 above)
Comment (alcxs) [reply Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: unified the lines, only the load library is different now depending on .net version

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public static class DbWorkarounds
+    {
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+        private static extern IntPtr LoadLibrary(string libname);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto)]
+        private static extern bool FreeLibrary(IntPtr hModule);
+
+#if NETCOREAPP
+        const string relativePath = @"\..\runtimes\win-x64\native\sni.dll";
```
Comment (ovidiuiliescu) [inline Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: Const should be NamedLikeThis.

Also, let's add the private modifier explicitly, and move it to the top of the class declaration.

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+#endif
+#if NETFRAMEWORK
+        const string relativePath = @"\..\runtimes\win-x86\native\sni.dll";
+#endif
+        public static void SNILoadWorkaround()
+        {
+
```
Comment (ovidiuiliescu) [inline Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: Empty line here can be removed.

**Review by ovidiuiliescu — CHANGES_REQUESTED** [review]

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs (hunk 3 above)
Comment (alcxs) [reply Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: modified

File: Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs (hunk 4 above)
Comment (alcxs) [reply Activities/Database/ConnectionDialog/ConnectionUIDialog/Workaround/DbWorkarounds.cs]: modified

**Review by ovidiuiliescu — APPROVED** [review]

**Pull request merged**